compiler/circle-mpqsolver/src/pattern/PatternSolver.cpp

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "PatternSolver.h"
+
+#include "PatternResolver.h"
+
+#include <iostream>
+
+using namespace mpqsolver::pattern;
+
+using LayerParam = luci::CircleQuantizer::Options::LayerParam;
+using LayerParams = luci::CircleQuantizer::Options::LayerParams;
+
+PatternSolver::PatternSolver(const std::string &input_quantization,
+                             const std::string &output_quantization,
+                             const std::vector<QuantizationPattern> &patterns)
+  : MPQSolver("", 1.f, input_quantization, output_quantization)
+{
+  MPQOptions options{patterns};
+  set_mpq_options(options);
+}
+
+std::unique_ptr<luci::Module> PatternSolver::run(const std::string &module_path)
+{
+  auto module = read_module(module_path);
+  if (!module)
+  {
+    throw std::runtime_error("Failed to load model");
```

Comment:
in `MPQSolver::read_module()`,
```cxx
  if (module.get() == nullptr)
  {
    std::cerr << "ERROR: Failed to load " << path << std::endl;
    return nullptr;
  }
```
which seems to show two `Failed to load` message.
IMHO, it maybe difficult to solve this issue in this PR so please fire a new issue for this problem, and go as-is in this PR.

Reply:
Ahh. Ok.